docs/extensions/tutorials/_posts/1970-01-05-Installing3rdPartyPackages.md

```diff
@@ -0,0 +1,330 @@
+
+<p class="image">
+<img src='{{ site.url }}/img/tutorials/installing-3rd-party-and-native-packages/swiper-in-preview.png'/>
+</p>
+
+## 2) Installing a Native Package
+
+A native package utilizes native capabilities of the underlying device. When installing such a package we have to make sure we link the native dependencies it has. This is done using [postinstall scripts](https://docs.npmjs.com/misc/scripts). As an example, we'll be making a QR Code reader that's going to display what the scanned QR code says. To scan a QR code we'll need to use the devices camera, which we'll get access to using `react-native-camera`, a 3rd party package for utilizing device cameras.
+
+### Making an Extension
+
+To begin, we'll need an extension, so let's just make one. If this is something you don't know how to do yet, you should really go through [My First Extension]({{ site.url }}/docs/extensions/my-first-extension/introduction) to make sure you understand the fundamentals.
+
+We begin by initializing an extension and writing in the basic information about it.
+
+```ShellSession
+$ shoutem init qr-reader-extension
+Enter information about your extension. Press `return` to accept (default) values.
+Title: QR Reader
+Version: 0.0.1
+Description: An extension that scans QR codes and displays the encoded information to the user.
+Initializing extension:
+Installing packages for server...
+...
+
+Extension initialized.
+```
+
+Let's switch over to the extension directory and add a screen with a shortcut that will be the user interface for our QR Reader.
+
+```ShellSession
+$ cd qr-reader-extension
+$ shoutem screen add QRReaderScreen --shortcut QRReader
+Enter shortcut information:
+Title: QR Reader
+
+Screen `QRReaderScreen` created in file `app/screens/QRReaderScreen.js`!
+Shortcut `QRReader` created.
+Shortcut `QRReader` opens `QRReaderScreen` screen.
+File `app/extension.js` was modified.
+File `extension.json` was modified.
+```
+
+### Installing the Package
+
+To make sure the native dependencies are linked, we'll have to make sure our custom postlink script is run by putting it in our `app/package.json` file. `rnpm`'s `postlink` command runs our `app/scripts/run.js` script that we'll explain afterwards.
+
+```json
+#file: app/package.json
+{
+  "name": "{{ site.example.devName }}.qr-reader-extension",
+  "version": "0.0.1",
+  "description": "An extension that scans QR codes and displays the encoded information to the user.",
+  "dependencies": {
+    "react-native-camera": "^0.4.1"
+  },
+  "rnpm": {
+    "commands": {
+      "postlink": "node node_modules/{{ site.example.devName }}.qr-reader-extension/scripts/run.js"
+    }
+  }
+}
+```
+
+> #### Note
+> `rnpm` refers to [React Native Package Manager](https://www.npmjs.com/package/rnpm), used for linking native dependancies in React Native apps.
```

Comment:
Explain why you used `tom.qr-reader-extension/scripts/run.js` which are inside of `node_modules`. Make a reference to [Technical overview](http://shoutem.github.io/docs/extensions/reference/overview) and explain that extensions are bundled inside of `node_modules`, which is why we run the script like that.